modules/light-clients/08-wasm/internal/ibcwasm/expected_interfaces.go

```diff
@@ -115,3 +115,21 @@ type WasmEngine interface {
 	// Unpin is idempotent.
 	Unpin(checksum wasmvm.Checksum) error
 }
+
+type WasmQuerier interface {
```

Comment:
I am fine having this interface, since we also added our own interface for `WasmEngine` even though there was [one available in `x/wasm`](https://github.com/CosmWasm/wasmd/blob/4442cc6df69caa2113a08d0e2e3151f0e462cf5f/x/wasm/types/wasmer_engine.go#L14). I think it's nice for consistency! :)

Reply:
The version that we initially implemented this feature for didn't have `WasmEngine` interface, that's why we went with using `wasmvm.Querier`. But now that `WasmEngine` is introduced, I am also in favor of having `WasmQuery` for the consistency. Added the assertion though :)

Reply:
I agree that if we want to go with `WasmQuerier`, it's better not to use `wasmvm.Querier` for the sake of consistency. But also important to note that now we need to do two additional tasks:
1. Maintain `WasmQuerier` interface
2. Consistently use `WasmQuerier` instead of `wasmvm.Querier`
I know that it's no big deal but these tiny nits adds additional development and reviewing costs since now the developers should be careful about using `WasmQuerier` instead of `wasmvm.Querier` and the reviewers should carefully check that `wasmvm.Querier` is not used. It might be a better idea to use `wasmvm.Querier` and drop `WasmQuerier` in this case. But I'd be happy to hear also your thoughts @crodriguezvega since @DimitrisJim is already in favor of this.

Reply:
Would like to note we added the `WasmEngine` interface because we did not want to have a direct dependency on `x/wasm`. If the wasmvm provided an engine interface, I suspect we would have used that directly. I would be in favour of using the wasmvm.Querier interface as I'm not sure I see the benefit of duplicating the interface?

Reply:
I think it would be good to reduce duplication of interfaces. As @colin-axner mentioned we added the `WasmEngine` interface because we didn't want to infer a circular dependency, as `x/wasm` depends on ibc-go.

I'm also in favour of using the `wasmvm.Querier` interface :)

Reply:
Sounds good to use `wasmvm.Querier` then. 👍